sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationproviderbase.py

```diff
@@ -0,0 +1,364 @@
+
+    if len(feature_filters_used) > 0:
+        filters_used = ""
+        if CUSTOM_FILTER_KEY in feature_filters_used:
+            filters_used = CUSTOM_FILTER_KEY
+        if PERCENTAGE_FILTER_KEY in feature_filters_used:
+            filters_used += ("+" if filters_used else "") + PERCENTAGE_FILTER_KEY
+        if TIME_WINDOW_FILTER_KEY in feature_filters_used:
+            filters_used += ("+" if filters_used else "") + TIME_WINDOW_FILTER_KEY
+        if TARGETING_FILTER_KEY in feature_filters_used:
+            filters_used += ("+" if filters_used else "") + TARGETING_FILTER_KEY
+        correlation_context += ",Filters=" + filters_used
+
+    correlation_context += _uses_feature_flags(uses_feature_flags)
+
+    if uses_key_vault:
+        correlation_context += ",UsesKeyVault"
+    host_type = ""
+    if AzureFunctionEnvironmentVariable in os.environ:
+        host_type = "AzureFunction"
+    elif AzureWebAppEnvironmentVariable in os.environ:
+        host_type = "AzureWebApp"
+    elif ContainerAppEnvironmentVariable in os.environ:
+        host_type = "ContainerApp"
+    elif KubernetesEnvironmentVariable in os.environ:
+        host_type = "Kubernetes"
+    elif ServiceFabricEnvironmentVariable in os.environ:
+        host_type = "ServiceFabric"
+    if host_type:
+        correlation_context += ",Host=" + host_type
+
+    if replica_count > 0:
+        correlation_context += ",ReplicaCount=" + str(replica_count)
+
+    if is_failover_request:
+        correlation_context += ",Failover"
+
+    if uses_load_balancing:
+        correlation_context += ",Features=LB"
+
+    headers["Correlation-Context"] = correlation_context
+    return headers
+
+
+def _uses_feature_flags(uses_feature_flags):
+    if not uses_feature_flags:
+        return ""
+    package_name = "featuremanagement"
+    try:
+        feature_management_version = version(package_name)
+        if feature_management_version:
+            return ",FMPyVer=" + feature_management_version
+    except PackageNotFoundError:
+        pass
+    return ""
+
+
+def is_json_content_type(content_type: str) -> bool:
+    if not content_type:
+        return False
+
+    content_type = content_type.strip().lower()
+    mime_type = content_type.split(";")[0].strip()
+
+    type_parts = mime_type.split("/")
+    if len(type_parts) != 2:
+        return False
+
+    (main_type, sub_type) = type_parts
+    if main_type != "application":
+        return False
+
+    sub_types = sub_type.split("+")
+    if "json" in sub_types:
+        return True
+
+    return False
+
+
+def _build_sentinel(setting: Union[str, Tuple[str, str]]) -> Tuple[str, str]:
+    try:
+        key, label = setting  # type:ignore
+    except IndexError:
+        key = setting
+        label = EMPTY_LABEL
+    if "*" in key or "*" in label:
+        raise ValueError("Wildcard key or label filters are not supported for refresh.")
+    return key, label
+
+
+def sdk_allowed_kwargs(kwargs):
```

Comment:
Is this new?

Reply:
Yes, the SDK throws an error if you pass an unknown kwarg into it. The SDK team suggested adding it. This came up in this PR as after I made the update the items used in the base class were not being popped from the kwargs any more due to the redesign and the SDK team suggested this as a solution.

Reply:
can sdk team provide this list or supported args? Instead of us having to manage it long term

Reply:
No, I had to get the list from the readme doc of azure core

Reply:
Discussed offline. SDK team suggested this approach for Python as they dont support any other option right now. We can revisit this when a better option is available.